### PR TITLE
[ESP32] Init OTA requestor in all-clusters-app

### PR DESCRIPTION
**Pull request by shubhamdp**
#### Problem
OTA requestor is not initialized in all-clusters-app/esp32.
Chip shell is using the weak function definitions for handling OTA commands.

#### Change overview
Initialize OTA requestor in all-clusters-app.
Also, enabled OTA_PROVIDER_CLIENT in all-cluster-app zap file.
Removed the weak function definitions from OTA command handlers.

#### Testing
Tested manually, all-clusters-app/esp32 and ota-provider-app/linux.
Triggered OTA using `matter ota query` chip shell command and
`./chip-tool otasoftwareupdaterequestor announce-ota-provider` chip-tool command.

### CODE REVIEW

**Review by wqx6 — APPROVED** [review]
Just one minor comment. LGTM otherwise.

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap [hunk 1]
```diff
@@ -1408,7 +1408,7 @@
           "mfgCode": null,
           "define": "OTA_PROVIDER_CLUSTER",
           "side": "server",
-          "enabled": 1,
+          "enabled": 0,
```
[the remark this answers is not in the document]
Comment (dhrishi) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:1411]: +1
IMO too, we should not disable any of the existing functionality

File: examples/all-clusters-app/esp32/main/CMakeLists.txt [hunk 2]
```diff
@@ -58,7 +58,7 @@ set(SRC_DIRS_LIST
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/low-power-server"
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/keypad-input-server"
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/media-playback-server"
-                      "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/ota-provider"
+                      "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/ota-requestor"
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/main/CMakeLists.txt:61]: And hence keep both provider and requestor here. Is it possible?

File: examples/all-clusters-app/esp32/main/main.cpp [hunk 3]
```diff
@@ -527,6 +541,18 @@ static void InitServer(intptr_t context)
...
+static void InitOTARequestor(void)
+{
+#if CONFIG_ENABLE_OTA_REQUESTOR
+    SetRequestorInstance(&gRequestorCore);
+    gRequestorCore.SetServerInstance(&Server::GetInstance());
+    gRequestorCore.SetOtaRequestorDriver(&gRequestorUser);
+    gImageProcessor.SetOTADownloader(&gDownloader);
+    gDownloader.SetImageProcessorDelegate(&gImageProcessor);
+    gRequestorCore.SetBDXDownloader(&gDownloader);
+#endif
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/main/main.cpp:553]: Too many ifdefs in the application? :-|

Comment (shubhamdp) [reply examples/all-clusters-app/esp32/main/main.cpp:549]: I removed the one which was guarding the header files.

File: examples/all-clusters-app/esp32/main/CMakeLists.txt (hunk 2 above)
Comment (shubhamdp) [reply examples/all-clusters-app/esp32/main/CMakeLists.txt:61]: Kept the ota-provider entry

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap (hunk 1 above)
[the remark this answers is not in the document]
Comment (shubhamdp) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.zap]: Reverted this change.

[the remark this answers is not in the document]
Comment (wqx6) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.zap]: https://github.com/project-chip/connectedhomeip/pull/12360#issuecomment-988037395 For current ZAP, it doesn't allow server and client of one cluster coexist on the same endpoint, so maybe we can try moving ota-provider client to endpoint 1.

**Review by selissia — APPROVED** [review]

**Review by dhrishi — APPROVED** [review]

**Comment by dhrishi** [comment]
@carol-apple PTAL

**Comment by dhrishi** [comment]
@shubhamdp What is the state of this? Is this dependent on any PR?

**Comment by shubhamdp** [comment]
> @shubhamdp What is the state of this? Is this dependent on any PR?

Waiting on CI

**Review by carol-apple — APPROVED** [review]

**Pull request merged**